### PR TITLE
Updating Maven task to use new locator server calls for authenticated feeds

### PR DESCRIPTION
**Pull request by kelliejos**
Updating the maven task to use the new locator server calls for authenticated feeds.

Tested on local test environment
Ran L0 tests
Ran legacy tests

### CODE REVIEW

File: Tasks/MavenV3/mavenutil.ts [hunk 1]
```diff
@@ -193,64 +188,73 @@ interface RepositoryInfo {
...
+            tl.debug(JSON.stringify(error));
+            packagingLocation = {
+                PackagingUris: [collectionUrl],
+                DefaultPackagingUri: collectionUrl};
+        }
+
+        let packageUrl = packagingLocation.DefaultPackagingUri;
+        tl.debug('collectionUrl=' + collectionUrl);
+        tl.debug('packageUrl=' + packageUrl);
+        let collectionHostname:string = url.parse(collectionUrl).hostname.toLowerCase();
```
Comment (lkillgore) [inline Tasks/MavenV3/mavenutil.ts]: Are we sure that it's right to use a `hostname` anymore? `https://dev.azure.com/kellie-test-project` would have the same hostname as `https://dev.azure.com/mseng`, correct?

Comment (kelliejos) [reply Tasks/MavenV3/mavenutil.ts]: Good point. I'll look into how I can make this more resilient.

File: Tasks/MavenV3/mavenutil.ts [hunk 2]
```diff
@@ -193,64 +188,78 @@ interface RepositoryInfo {
     id:string;
 }
 
-function collectFeedRepositories(pomContents:string): Q.Promise<any> {
-    return convertXmlStringToJson(pomContents).then(function (pomJson) {
+async function collectFeedRepositories(pomContents:string): Promise<any> {
+    return convertXmlStringToJson(pomContents).then(async function (pomJson) {
```
Comment (yacaovsnc) [inline Tasks/MavenV3/mavenutil.ts]: Seems we are mixing async and Promise style together.  Maybe cleaner to just await for the `convertXmlStringToJson`, and write it as imperative style.

File: Tasks/MavenV3/mavenutil.ts [hunk 3]
```diff
@@ -193,64 +188,78 @@ interface RepositoryInfo {
...
+        const collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
+        let packagingLocation: pkgLocationUtils.PackagingLocation;
+        try {
+            packagingLocation = await pkgLocationUtils.getPackagingUris(pkgLocationUtils.ProtocolType.Maven);
+        } catch (error) {
+            tl.debug("Unable to get packaging URIs, using default collection URI");
+            tl.debug(JSON.stringify(error));
+            packagingLocation = {
+                PackagingUris: [collectionUrl],
+                DefaultPackagingUri: collectionUrl};
```
Comment (yacaovsnc) [inline Tasks/MavenV3/mavenutil.ts]: nit: curly brace should be on a new line.

File: Tasks/MavenV3/mavenutil.ts [hunk 4]
```diff
@@ -193,64 +188,78 @@ interface RepositoryInfo {
...
+        }
+
+        let packageUrl = packagingLocation.DefaultPackagingUri;
+        tl.debug('collectionUrl=' + collectionUrl);
+        tl.debug('packageUrl=' + packageUrl);
+        let collectionName:string = url.parse(collectionUrl).hostname.toLowerCase();
+        let collectionPathName = url.parse(collectionUrl).pathname;
+        if(collectionPathName && collectionPathName.length > 1) {
```
Comment (yacaovsnc) [inline Tasks/MavenV3/mavenutil.ts]: nit: space after `if`

**Pull request merged**